src/Entities/QueryApi/DataPlatformIndexerStatuses.jsx

```diff
@@ -3,31 +3,68 @@ const GRAPHQL_ENDPOINT = props.GRAPHQL_ENDPOINT || "https://near-queryapi.api.pa
 const indexerAccount = props.indexerAccount || "dataplatform.near";
 const indexerAccountLink = indexerAccount.replace(".", "_");
 const indexerFilter = props.indexerFilter || null;
-const fullFilter = indexerFilter ? indexerAccount + "/" + indexerFilter : indexerAccount;
 
-const [statuses, setIndexerStatuses] = useState([]);
+const [statuses, setIndexerStatuses] = useState({});
 const [errors, setErrors] = useState("");
 const [timer, setTimer] = useState(null);
+
+const [latestBlock, setLatestBlock] = useState(0);
+const [indexerList, setIndexerList] = useState(null);
+const registryContract = "queryapi.dataplatform.near";
+const registry = Near.view(registryContract, "list_indexer_functions", {
+  account_id: indexerAccount,
+});
+
+if (!registry) {
+  return <div>Loading indexer list from contract...</div>;
+} else {
+  try {
+    setIndexerList(Object.keys(registry["Account"]).sort());
+  } catch (e) {
+    setErrors(e);
+  }
+}
+
+const defaultIndexerList = [
+  "access_keys_v1",
+  "accounts",
+  "components",
+  "entities",
+  "feed",
+  "moderation",
+  "notifications",
+  "social_feed",
+  "verifications",
+];
+
+if (!indexerList) {
+  return (
+    <div>
+      <p>Indexer list failed to load from contract</p>
+      <button onClick={() => setIndexerList(defaultIndexerList)}>Use default indexer list</button>
+    </div>
+  );
+}
+
 function fetchGraphQL(operationsDoc, operationName, variables) {
   return asyncFetch(`${GRAPHQL_ENDPOINT}/v1/graphql`, {
     method: "POST",
-    headers: { "x-hasura-role": "append" },
+    headers: { "x-hasura-role": indexerAccountLink },
     body: JSON.stringify({
       query: operationsDoc,
       variables: variables,
       operationName: operationName,
     }),
   });
 }
-const query = `query MyQuery {
-  indexer_state(where: {function_name: {_like: "${fullFilter}%"}}) {
-    status
-    current_block_height
-    function_name
+const query = (indexer) => `query MyQuery {
+  ${indexerAccountLink}_${indexer}_sys_metadata {
```

Comment:
I know none of the current dataplatform indexers have this problem, but the graphQL query may not work if it has symbols like hyphens, which are allowed in the indexer name but not the schema name. I'd run the returned names through some sanitization before putting them in the query. Here's how we do it currently:
```
const sanitizedAccountId = indexerDetails.accountId.replace(/[^a-zA-Z0-9]/g, '_').replace(/^([0-9])/, '_$1');
  const sanitizedIndexerName = indexerDetails.indexerName.replace(/[^a-zA-Z0-9]/g, '_').replace(/^([0-9])/, '_$1');
  const functionName = `${indexerDetails.accountId}/${indexerDetails.indexerName}`;
  const schemaName = `${sanitizedAccountId}_${sanitizedIndexerName}`;
  const tableName = `${schemaName}_sys_logs`;
```